doc/book/in-depth-guide/models-and-servicemanager.md

```diff
@@ -596,9 +595,9 @@ template to display the data.
 When pushing variables to the view, they are accessible in two ways: either
 using object notation (`$this->posts`) or implicitly as script-level variables
 `$posts`). The two approaches are equivalent; however, calling `$posts` results
-in a little round-trip through the renderer's `__get()` method. We often
-recommend using `$this` notation to visually differentiate between variables
-passed to the view, and those created within the script itself.
+in a little round-trip through the renderer's `__get()` method <<what renderer? Hasn’t been mentioned yet. Also, doesn’t seem much point in mentioning this unless you explain why it matters.>>. We often
+recommend using `$this` notation <<is '$this notation' the same thing as object notation mentioned two sentences above? If so, can this be made clear to avoid confusion?>> to visually differentiate between variables
+passed to the view, and those created within the script itself. <<It’s not clear to me what the difference is between these two types of variables. I think it would be helpful if it was explained, even if just in a linked-to side note.>>
```

Comment:
Probably a better way to differentiate is as _instance variables_ versus _script variables_.

Instance variables are referenced using `$this->{variable name}` (e.g., `$this->posts`), while script variables are referenced using standard variable notation (`${variable name}`, e.g., `$posts`).

By default, we import all variables passed via a view model to the renderer directly into the view script, and you can thus reference them as either instance or script variables (i.e., `$this->posts` is the same as `$posts`). Our recommendation is to reference any variables defined as part of the original view model using _instance variable_ notation, to make it clear where they originate, and to only use _script variable_ notation for variables defined directly in the script itself.

In terms of this latter, consider loops, for example:

```php
<?php foreach ($this->posts as $post) : ?>
```

Here, we know that `$this->posts` references `posts` variable provided in the view model. `$post` is a single iteration as dereferenced in the script itself.

Other times, we might initialize a variable in our view script; for example, a date:

```php
<?php
$curDate = new DateTime('now');
```

Other times, we might process _instance variables_ to create view-specific artifacts:

```php
$posts = array_filter($this->posts, function ($post) use ($curDate) {
    return $post->published <= $curDate;
});
```

So, my take on your comment:

- I think we need a block quote to give us the _instance variable_ versus _script variable_ terminology.
- We likely need examples such as those I've given above to demonstrate the differences in usage.
- The information on `__get()` likely needs an additional sentence, explaining that the fact that view model variables are proxied via `__get()` means no introspection by your IDE.

Reply:
I've added a block quote covering instance variable versus script variable usage (based on your explanation) which makes sense to me. I haven't included the code examples above as they didn't seem necessary (although they helped me to understand). I think they would break the flow too much and, if required, it would be better to have them in a separate topic and provide a link in the tutorial.

I opted for removing the reference to __get, as, to be honest, I'm still not clear quite how it works and it just broke the flow for me. If it's an important point to make about the IDE not providing any hints for variables, I think it should be note / block quote so it's clear that it's a separate issue.